frontend/pages/collections/[slug].page.tsx

```diff
@@ -78,3 +103,113 @@ export const getStaticPaths: GetStaticPaths = async () => {
     fallback: "blocking",
   };
 };
+
+const MobileViewComponent: FC<CollectionProps> = ({ collection }: CollectionProps) => {
+  const [showIcons, setShowIcons] = useState(false);
+  const onClick = () => {
+    setShowIcons(true);
+    if (showIcons) {
+      setShowIcons(false);
+    }
+  };
+  return (
+    <>
+      <Container>
+        <Grid sx={{ m: 3, display: "flex", alignItems: "center", justifyContent: "flex-end" }}>
+          <Grid sx={{ maxWidth: 250, alignItems: "center", justifyContent: "center" }}>
+            <Grid
+              sx={{
+                flexDirection: "column",
+                display: "flex",
+                alignItems: "center",
+                justifyContent: "center",
+              }}
+            >
+              <Button
+                variant="contained"
+                color="primary"
+                onClick={onClick}
+                sx={{ borderRadius: "50%" }}
+              >
+                <ShareIcon />
+              </Button>
+              <>
+                {showIcons ? (
+                  <Grid
+                    container
+                    spacing={2}
+                    alignItems="center"
+                    justifyContent="center"
+                    sx={{ my: 1 }}
+                  >
+                    <Grid item>
+                      <FacebookShareButton
+                        url={`https://modularhistory.com/collections/${collection.slug}`}
+                        quote={collection.title}
+                      >
+                        <FacebookIcon size={36} round={true} />
+                      </FacebookShareButton>
+                    </Grid>
+                    <Grid item>
+                      <TwitterShareButton
+                        url={`https://modularhistory.com/collections/${collection.slug}`}
+                        title={collection.title}
+                      >
+                        <TwitterIcon size={36} round={true} />
+                      </TwitterShareButton>
+                    </Grid>
+                    <Grid item>
+                      <EmailShareButton
+                        url={`https://modularhistory.com/collections/${collection.slug}`}
+                        subject={collection.title}
+                      >
+                        <EmailIcon size={36} round={true} />
+                      </EmailShareButton>
+                    </Grid>
+                  </Grid>
+                ) : null}
+              </>
+            </Grid>
+          </Grid>
+        </Grid>
+      </Container>
+    </>
+  );
+};
+
+const DesktopViewComponent: FC<CollectionProps> = ({ collection }: CollectionProps) => {
+  return (
+    <Container>
+      <Grid sx={{ m: 3, display: "flex", alignItems: "center", justifyContent: "flex-end" }}>
+        <Grid sx={{ maxWidth: 250, alignItems: "center", justifyContent: "center" }}>
+          <Grid container spacing={2} alignItems="center" justifyContent="center">
```

Comment:
This level of nested components is almost certainly unnecessary. @sunzae99 can you experiment with removing props/grids and see if this can be reduced?

Reply:
+1, let's use just a single `<Grid container ... >` to wrap these grid items.

We also shouldn't need the `<Container>`. If it adds padding to the grid, then if necessary, we can just add the padding to the grid `sx` instead.

Reply:
> we can just add the padding to the grid via `sx` instead.

We actually don't even need `sx`. https://mui.com/api/grid/:
> As a CSS utility, the Grid component also supports all system properties. You can use them as props directly on the component.